resdk/resources/base.py

```diff
@@ -87,19 +91,49 @@ def __init__(self, slug=None, id=None,  # pylint: disable=redefined-builtin
 
         self.logger = logging.getLogger(__name__)
 
-    def _update_fields(self, fields):
+    def fields(self):
+        return self.WRITABLE_FIELDS + self.UPDATE_PROTECTED_FIELDS + self.READ_ONLY_FIELDS
+
+    def _update_fields(self, payload):
         """Update resource fields.
 
-        :param fields: Resource fields
-        :type fields: dict
+        :param payload: Resource fields
+        :type payload: dict
 
         """
-        for field_name, field_value in fields.items():
-            setattr(self, field_name, field_value)
+        self._original_values = payload
+        for field_name in self.fields():
+            setattr(self, field_name, payload.get(field_name, None))
 
     def update(self):
         """Update resource fields from the server."""
-        self._update_fields(self.api(self.id).get())
+        response = self.api(self.id).get()
+        self._update_fields(response)
+
+    def save(self):
+        """Save resource to the server."""
+        if self.id:  # update resource
+            payload = {}
+            for field_name in self.WRITABLE_FIELDS:
```

Comment:
Yes, I think that if a field is not writable and the user attempts to change it, it should be rejected.

> I would have to somehow distinguish the changes from the server and users.

You can track dirty fields by using [descriptors](https://docs.python.org/2.7/howto/descriptor.html) for them, as long as they are scalars (eg. not lists/dicts), set using operator `=` or `setattr`. If you want to track modifications to non-scalar fields, you need some additional logic (a dirty-tracking list/dict/etc.).